### PR TITLE
Add linter; auto-lint for staged files on commit

### PR DESCRIPTION
**Pull request by cirrahn**
Ensure a consistent code style for JS/JSON files. This avoids messy diffs, and allows us to confidently run mass-edit scripts on pack macros.

This is largely borrowed from the Lancer system's setup, see:

- https://github.com/Eranziel/foundryvtt-lancer/pull/470/files
- https://github.com/Eranziel/foundryvtt-lancer/blob/master/.prettierrc.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,12 +8,17 @@
         "db:pack": "node tooling/pack-macros.mjs",
         "new-effect": "node tooling/create-new-effect.mjs",
         "test": "npm run test:js",
-        "test:js": "node --experimental-vm-modules node_modules/jest/bin/jest.js"
+        "test:js": "node --experimental-vm-modules node_modules/jest/bin/jest.js",
+        "format": "prettier --write scripts packs_source lang tooling test",
+        "prepare": "npx husky init"
```
Comment (cirrahn) [inline package.json:13]: `"prepare"` sets up `husky` whenever a bare `npm install` is run. See: https://docs.npmjs.com/cli/v10/using-npm/scripts#life-cycle-scripts

**Comment by Z3nner** [comment]
I'm not a huge fan of how it's formatting the macros.
I think I'd prefer the following be indented by one less level if that's not a problem.
.effect()
.sound()
.canvasPan()
.play()

**Comment by cirrahn** [comment]
Looks like that would be a problem! `prettier` does not have any concept of a "post-processing" step where the extra formatting could be applied. Although, I might be able to break apart the standard JavaScript parser plugin and add some custom logic, but I will have to investigate as it looks non-trivial to do so.

Will update or close the PR if/when I figure it out!

**Comment by dodgepong** [comment]
As somebody who has gone through a full-codebase-formatting experience before, I know there are occasionally some paint points, but I do think the value of enforcing consistent formatting is worth the small nits of how the auto-formatter can somethings format things slightly weirdly.

File: .lintstagedrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+	"scripts/**/*": "prettier --write",
+	"packs_source/**/*": "node tooling/prettier/prettier-packs.js",
```
Comment (cirrahn) [inline .lintstagedrc:3]: Added a custom wrapper for files in `packs_source`. They get the standard `prettier` treatment, but then an additional post-processing step to re-format some lines.

File: tooling/prettier/prettier-packs.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import fs from "fs";
```
Comment (cirrahn) [inline tooling/prettier/prettier-packs.js:1]: I spent a few hours trying to hack into the guts of `prettier`, but it was looking like I would have to restructure an unfamiliar AST. So, I gave up and made this simple post-processor script instead!

**Comment by cirrahn** [comment]
@Z3nner updated to de-indent your chosen lines. I am not sure I hit the intended effect, though. We might want to instead indent everything which is _not_ in your list of lines? For example, see: https://github.com/Z3nner/lancer-weapon-fx/pull/41/files#diff-92654fc0740600df32b845cd50fbca21db9be6ee4fb6f091e3d4025c0f3ecdd9R13

**Comment by Z3nner** [comment]
You're probably right, about indenting more. Hate to nit pick, my formatting is far from perfect but it's gotten better and I just didn't want to lose all that. I see the advantage for the rest of the module for sure, but from my perspective and work environment the original method would have been a step backwards. Either method you've worked up or proposed is good with me.

**Comment by cirrahn** [comment]
@Z3nner take two -- flipped to over-indent lines *not* in your list, when they are part of a `sequencer. ...` call chain. Better?

**Comment by Z3nner** [comment]
Yes! That looks great!

**Pull request merged**